src/storage/s3.rs

```diff
@@ -132,6 +182,61 @@ impl<'a> StorageTransaction for S3StorageTransaction<'a> {
         })
     }
 
+    fn delete_prefix(&mut self, prefix: &str) -> Result<(), Error> {
+        self.s3.runtime.handle().block_on(async {
+            let mut continuation_token = None;
+            loop {
+                let list = self
+                    .s3
+                    .client
+                    .list_objects_v2(ListObjectsV2Request {
+                        bucket: self.s3.bucket.clone(),
+                        prefix: Some(prefix.into()),
+                        continuation_token,
+                        ..ListObjectsV2Request::default()
+                    })
+                    .await?;
+
+                let to_delete = list
+                    .contents
+                    .unwrap_or_else(Vec::new)
+                    .into_iter()
+                    .filter_map(|o| o.key)
+                    .map(|key| ObjectIdentifier {
+                        key,
+                        version_id: None,
+                    })
+                    .collect::<Vec<_>>();
+
+                let resp = self
+                    .s3
+                    .client
+                    .delete_objects(DeleteObjectsRequest {
+                        bucket: self.s3.bucket.clone(),
+                        delete: rusoto_s3::Delete {
+                            objects: to_delete,
+                            quiet: None,
+                        },
+                        ..DeleteObjectsRequest::default()
+                    })
+                    .await?;
+
+                if let Some(errs) = resp.errors {
+                    for err in &errs {
+                        log::error!("error deleting file from s3: {:?}", err);
+                    }
+
+                    failure::bail!("uploading to s3 failed");
+                }
+
+                continuation_token = list.continuation_token;
+                if continuation_token.is_none() {
```

Comment:
Is the `continuation_token` temp var even needed here? It looks like you could just use
```rust
if list.continuation_token.is_none() {
    return Ok(());
}
```

Reply:
The variable is used in the next iteration to continue with the listing, so yes, it's needed.